python/delta_sharing/reader.py

```diff
@@ -368,12 +372,15 @@ def __table_changes_to_pandas_kernel(self, cdfOptions: CdfOptions) -> pd.DataFra
                 table, interface, min_version, max_version
             ).build()
 
+            scan_result = scan.execute(interface)
             if num_versions_with_action == 0:
-                schema = scan.execute(interface).schema
+                schema = scan_result.schema
                 result = pd.DataFrame(columns=schema.names)
+            elif self._convert_in_batches:
```

Comment:
do you think it's worth adding some logging on the batch behavior for debugging/verification purpose?